torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -998,3 +1006,554 @@ def test_call_statement(
             for arg_type, arg_name in zip(arg_types, arg_names)
         ]
         return f"{kernel.kernel_name}({', '.join(arguments)}, workspace_size_ptr, (uint8_t*)workspace_data.get(), 0);"
+
+
+@dataclass
+class CKGemmOperation:
```

Comment:
I think, here as well, we should not pretend that CUDA and ROCm are the same. They should have separate packages ( torch._inductor.codegen.rocm or such ) with compatible interfaces where possible instead of trying to make the CUDA specific codebase also handle ROCm. 

Otherwise it gets very hard to maintain either, especially  considering that most people who develop on the CUDA side don't have the ability to test and debug locally for ROCm.

Reply:
Agreed, these seem like they should be part of their own package